### ISSUE TITLE
why weight decay in _loss function is only applied to the first layer

### ISSUE THREAD

**Issue by ZhanqiLiu** [google-research/google-research#95]
def _loss(self):
    """Construct the loss function."""
    # Weight decay loss
    if FLAGS.weight_decay > 0.0:
      for var in self.layers[0].vars.values():
        self.loss += FLAGS.weight_decay * tf.nn.l2_loss(var)
